### PR TITLE
move Downloader&zipUtils namespace from jsb to native & refine doc.

### PR DESCRIPTION
**Pull request by novain1102**
Re: https://github.com/cocos/cocos-engine/issues/11135

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: @types/jsb.d.ts [hunk 1]
```diff
@@ -193,29 +193,47 @@ declare namespace jsb {
          * ```ts
          * // create a download task
          * let downloader = new jsb.Downloader();
-         * let task = downloader.createDownloadTask('https://example.com/exampleFile.zip', jsb.fileUtils.getWritablePath());
-         * down.setOnTaskProgress((task, recv, total)=>{ console.log(recv, total);})
+         * let task = downloader.createDownloadFileTask('https://example.com/exampleFile.zip', jsb.fileUtils.getWritablePath());
          */
         createDownloadFileTask (requestURL:string, storagePath:string, identifier?:string): DownloaderTask;
```
Comment (minggo) [inline @types/jsb.d.ts]: Can remove Downloader related docs here, as it is moved to `cocos/native-binding/index.ts`.

**Comment by suqionglian** [comment]
@cocos-robot run test cases

File: @types/jsb.d.ts (hunk 1 above)
Comment (novain1102) [reply @types/jsb.d.ts]: So remove docs and just keep APIs in jsb.d.ts ?

Comment (minggo) [reply @types/jsb.d.ts]: jsb.d.ts just keep the API that are not used by developers.

Comment (PPpro) [reply @types/jsb.d.ts]: if the API exported to user, it should put in the native-bindinng module

if the API is only for engine dev, it should put in the jsb.d.ts

the declaration should be merged together in the future, here is the trackinng issue https://github.com/cocos/3d-tasks/issues/12609

Comment (novain1102) [reply @types/jsb.d.ts]: > jsb.d.ts just keep the API that are not used by developers.

Already removed.

Comment (novain1102) [reply @types/jsb.d.ts]: > if the API exported to user, it should put in the native-bindinng module
> 
> if the API is only for engine dev, it should put in the jsb.d.ts
> 
> the declaration should be merged together in the future, here is the trackinng issue [cocos/3d-tasks#12609](https://github.com/cocos/3d-tasks/issues/12609)

got it.

**Review by minggo — APPROVED** [review]

**Review by PatriceJiang — APPROVED** [review]

**Pull request merged**